arxiv/vault/tests/test_manager.py

```diff
@@ -0,0 +1,229 @@
+"""Tests for :mod:`.manager`."""
+
+from unittest import TestCase, mock
+from datetime import datetime
+import time
+from pytz import UTC
+
+from .. import manager
+from ..core import Secret
+
+
+class TestGetSecretsNotAuthenticated(TestCase):
+    """We use a :class:`.SecretsManager` to grab Vault secrets."""
+
+    def setUp(self):
+        """We have a :class:`.Vault` connection and are not authenticated."""
+        self.vault = mock.MagicMock(authenticated=False)
+
+    def test_generic_request(self):
+        """The app requires a generic secret."""
+        requests = [
+            manager.SecretRequest.factory('generic', **{
+                'name': 'GENERIC_FOO',
+                'path': 'baz',
+                'key': 'foo',
+                'mount_point': 'foo/'
+            })
+        ]
+        self.vault.generic.return_value = Secret('foosecret',
+                                                 datetime.now(UTC),
+                                                 'foolease-1234', 1234, True)
+        secrets = manager.SecretsManager(self.vault, requests)
+
+        yields = {k: v for k, v in secrets.yield_secrets('tôken', 'röle')}
```

Comment:
tôken and röle seem to be unused parameters. What would these parameters be when used non-mocked?

Reply:
Added this to the ``yield_secrets`` docstring:

```
        Parameters
        ----------
        token : str
            Token for authenticating with Vault. For example, the Kubernetes
            ServiceAccount token used to authenticate with the Kubernetes
            auth method.
        role : str
            The name of the Vault role associated with the token.
```